### PR TITLE
std: Remove some assumptions about the host platform

### PR DESCRIPTION
**Pull request by LemonBoy**
The stdlib is now 32-bit friendly.

### CODE REVIEW

File: std/crypto/chacha20.zig [hunk 1]
```diff
@@ -161,7 +161,8 @@ pub fn chaCha20With64BitNonce(out: []u8, in: []const u8, counter: u64, key: [32]
     c[3] = mem.readIntSliceLittle(u32, nonce[4..8]);
 
     const block_size = (1 << 6);
-    const big_block = (block_size << 32);
+    // The full block size is greater than the address space on a 32bit machine
+    const big_block = if (comptime @sizeOf(usize) > 4) (block_size << 32) else maxInt(usize);
```
Comment (andrewrk) [inline std/crypto/chacha20.zig]: this `comptime` is redundant

File: std/os/time.zig [hunk 2]
```diff
@@ -31,8 +31,8 @@ pub fn sleep(nanoseconds: u64) void {
 
 pub fn posixSleep(seconds: u63, nanoseconds: u63) void {
     var req = posix.timespec{
-        .tv_sec = seconds,
-        .tv_nsec = nanoseconds,
+        .tv_sec = @intCast(isize, seconds),
```
Comment (andrewrk) [inline std/os/time.zig]: This introduces a change to the API of this function: previously any value that fit inside of a u63 was allowed. After this change, the integer range between pointer size and u63 is an assertion failure.

I think it's good that the sleep function cannot fail. I also think it should have legal behavior for the entire input range. Here is my proposal to accomplish this:
 * Add documentation comment to note that posixSleep() may have spurious wakeups. This is already the case, as noted by the EINVAL case.
 * If the number of seconds/nanoseconds exceeds the capacity of the integers in the `timespec` struct, do the sleep syscall for the maximum duration, and then simply return, treating it as a spurious wakeup.

Comment (LemonBoy) [reply std/os/time.zig]: I've implemented another solution, we just slice and dice the number of seconds and do N calls to `nanosleep` until we've slept for the desired amount of time.

Comment (andrewrk) [reply std/os/time.zig]: Ah I'm sorry I didn't see this comment until later, and I found the new code harder to reason about and verify than my idea and so I modified the logic when merging (I also added doc comments and fixed the other sleep function). If you feel strongly about it and are willing to put up another PR I'll merge it.

Comment (LemonBoy) [reply std/os/time.zig]: > If you feel strongly about it and are willing to put up another PR I'll merge it.

Nevermind, sleeping for such a long time is quite a niche use-case that any behavior (as long as it's documented) is fine.

**Pull request merged**